internal/rbd/rbd_util.go

```diff
@@ -1733,3 +1734,41 @@ func (rs *rbdSnapshot) isCompatibleThickProvision(dst *rbdVolume) error {
 
 	return nil
 }
+
+func (ri *rbdImage) addSnapshotScheduling(
+	interval admin.Interval,
+	startTime admin.StartTime) error {
+	ls := admin.NewLevelSpec(ri.Pool, ri.RadosNamespace, ri.RbdImageName)
+	ra, err := ri.conn.GetRBDAdmin()
+	if err != nil {
+		return err
+	}
+	adminConn := ra.MirrorSnashotSchedule()
+	// list all the snapshot scheduling and check at least one image scheduling
+	// exists with specified interval.
+	ssList, err := adminConn.List(ls)
+	if err != nil {
+		return err
+	}
+
+	for _, ss := range ssList {
+		// make sure we are matching image level scheduling. The
+		// `adminConn.List` lists the global level scheduling also.
+		if ss.Name == ri.String() {
+			for _, s := range ss.Schedule {
+				//  TODO Add support to check start time also.
```

Comment:
nit:
I think below format is good. 
```
// TODO: ... 
```

Reply:
@Madhu-1 just to understand this: so in general setup, do we expect different `ss` exist for different images where mirroring enabled ? or a group kind of `ss` can exist ? from the code, it looks like former is the case though.  Also, do we have a provision to check image specific `ss` via go ceph or we have to iterate over the complete list and map the image name and schedule name?  because here in this function, we are retrieving the list and then checking `its` name matches with what we have in hand via `ri.image`, so the question.

Reply:
Done

Reply:
@Madhu-1 https://github.com/ceph/ceph-csi/pull/2216#discussion_r664360483

Reply:
the scheduling can be defined at the pool level (this is not under cephcsi control) and the image level. when we call `List` it returns all different level schedules defined on the image, not for all images. we are doing the comparison here to make sure the image scheduling exists with the specified internal.

Reply:
@Madhu-1 if first check (`ss.Name =ri.Image`) satisfy (regardless of `pool/namespace/image` level) we validate the  `interval`, I believe its fine with the design and we dont need to check any other scdhules in this case. Isnt it ?

Reply:
Yes for now we need to validate interval and a todo is added to validate start time also

Reply:
ok. Thanks for the confirmation